pkg/utils/foreignCluster/getForeignCluster.go

```diff
@@ -48,3 +55,129 @@ func GetOlderForeignCluster(
 	}
 	return foreignCluster
 }
+
+// getAuthAddress the external address where the Authentication Service is reachable from the external world.
+func getAuthAddress(ctx context.Context, namespacedClient, clusterWideClient client.Client, authServiceAddress, namespace string) (string, error) {
+	// this address can be overwritten setting this environment variable
+	address := authServiceAddress
+	if address != "" {
+		return address, nil
+	}
+
+	// get the authentication service  (the namespace is automatically inferred by the namespaced client).
+	var svc corev1.Service
+	ref := types.NamespacedName{Name: liqoconst.AuthServiceName, Namespace: namespace}
+	if err := namespacedClient.Get(ctx, ref, &svc); err != nil {
+		klog.Error(err)
+		return "", err
+	}
+
+	// if the service is exposed as LoadBalancer
+	if svc.Spec.Type == corev1.ServiceTypeLoadBalancer {
+		// get the IP from the LoadBalancer service
+		if len(svc.Status.LoadBalancer.Ingress) == 0 {
+			// the service has no external IPs
+			err := goerrors.New("no valid external IP for LoadBalancer Service")
+			klog.Error(err)
+			return "", err
+		}
+		lbIngress := svc.Status.LoadBalancer.Ingress[0]
+		// return the external service IP
+		if hostname := lbIngress.Hostname; hostname != "" {
+			return hostname, nil
+		} else if ip := lbIngress.IP; ip != "" {
+			return ip, nil
+		} else {
+			// the service has no external IPs
+			err := goerrors.New("no valid external IP for LoadBalancer Service")
+			klog.Error(err)
+			return "", err
+		}
+	}
+
+	// only physical nodes
+	//
+	// we need to get an address from a physical node, if we have established peerings in the past with other clusters,
+	// we may have some virtual nodes in our cluster. Since their IPs will not be reachable from other clusters, we cannot use them
+	// as address for a local NodePort Service
+	req, err := labels.NewRequirement(liqoconst.TypeLabel, selection.NotIn, []string{liqoconst.TypeNode})
+	utilruntime.Must(err)
+
+	// get the IP from the Nodes, to be used with NodePort services
+	nodes := corev1.NodeList{}
+	if err := clusterWideClient.List(ctx, &nodes, client.MatchingLabelsSelector{Selector: labels.NewSelector().Add(*req)}); err != nil {
```

Comment:
Would double-check, but the namespaced client should be able to access cluster-wide resources (if that makes sense to reduce the number of parameters)